libraries/rush-lib/src/logic/base/BaseInstallManager.ts

```diff
@@ -160,16 +163,23 @@ export abstract class BaseInstallManager {
       subspace,
       { npmrcHash: npmrcHash || '<NO NPMRC>' }
     );
+    if (isFilteredInstall) {
+      // Get the projects involved in this filtered install
+      const selectedProjectNames: string[] = this.options.pnpmFilterArguments.filter(
+        (filterArg) => filterArg !== '--filter'
+      );
```

Comment:
It also won't match if `project-b` depends on `project-a` and you do `rush install --to project-b` and then `rush install --to project-a`.

Reply:
This has been refactored to not filter out "--filter" anymore, only passing the package names directly in the options.

> It also won't match if project-b depends on project-a and you do rush install --to project-b and then rush install --to project-a.

In this scenario, since the selectedProjectNames subset for installing "--to project-b" is larger than the union of "--to project-b" + "--to project-a", then it should also skip the install

Reply:
Okay this has once again been refactored to just use the selected projects directly, instead of the filter arguments so I think this issue is resolved